pravega/pravega
# Issues 4883, 5013: Upgrade grpc-netty to 1.31.0

**Change log description**  
Netty was only partially upgraded in #4851. This PR upgrades Netty as used by gRPC in client/server communications.

**Purpose of the change**  
Fixes #4883, fixes #5013

**What the code does**  
Accommodates a breaking change in setting up load balancing for ManagedChannelBuilder. Applies `@SuppressWarnings("deprecation")` to methods using the `nameResolverFactory()` API in order to bypass warnings given our use of `-Werror`.

**How to verify it**  
See it working here: https://github.com/derekm/quarkus-pravega-webhooks

This change will allow pravega-client to be brought into modern Java microservices frameworks such as Quarkus.io (uses Netty 4.1.49.Final), Micronaut.io (uses Netty 4.1.48.Final), Vert.x (uses Netty 4.1.49.Final), Riposte (uses Netty 4.1.49.Final), etc.

Spark 3.x uses at least Netty 4.1.47.Final. (But Spark 2.4.x uses a combination of Netty 3.9.9.Final & 4.1.42.Final.)
